crypto/fipsmodule/rsa/rsa_impl.c

```diff
@@ -113,6 +108,42 @@ int rsa_check_public_key(const RSA *rsa) {
     OPENSSL_PUT_ERROR(RSA, RSA_R_KEY_SIZE_TOO_SMALL);
     return 0;
   }
+
+  return 1;
+}
+
+rsa_key_type_t rsa_detect_key_type(const RSA *rsa) {
```

Comment:
This could be a static function in `rsa.c`.

Reply:
sure.